pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/settings/GetFieldSettingsRoute.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.dsm.route.settings;
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.FieldSettings;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.statics.UserErrorMessages;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class GetFieldSettingsRoute extends RequestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(GetFieldSettingsRoute.class);
+
+    @Override
+    public Object processRequest(Request request, Response response, String userId) throws Exception {
+        String realm = request.params(RequestParameter.REALM);
+        if (StringUtils.isBlank(realm)) {
+            throw new IllegalArgumentException("Realm cannot be empty");
+        }
+
+        String userIdRequest = UserUtil.getUserId(request);
+        if (RoutePath.RequestMethod.GET.toString().equals(request.requestMethod())) {
```

Comment:
I am getting bad feeling about checking whether request method is `GET` that in future this class may get very ugly, that someone will follow this pattern and use this class for `POST` method and we will dive into ugly, stair like `if-else` statements.